Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,488 @@
+  · ext x; rw [pow_two]; exact (abs_mul_abs_self _).symm
+  · exact tendsto_norm_cocompact_atTop.atTop_mul_atTop tendsto_norm_cocompact_atTop
+
+lemma cocompact_strip.tendsto_sq_im_atTop : Tendsto (fun z : ℂ ↦ z.im^2)
+    (cocompactStrip 0 1) atTop := by
+  apply tendsto_inf_left _
+  change Tendsto ((fun x ↦ x^2) ∘ im) (comap im (cocompact ℝ)) atTop
+  rw [tendsto_comap'_iff]
+  · convert_to Tendsto (fun x ↦ ‖x‖ * ‖x‖) (cocompact ℝ) atTop
+    · ext x; simp only [Real.rpow_two, pow_two, Real.norm_eq_abs, abs_mul_abs_self]
+    · exact tendsto_norm_cocompact_atTop.atTop_mul_atTop tendsto_norm_cocompact_atTop
+  ·  simp only [univ_mem, range_im]
+
+lemma cocompact_strip.eventually_sq_re_le_one : ∀ᶠ (z : ℂ) in
+    (cocompactStrip 0 1), z.re^2 ≤ 1 := by
+  rw [cocompactStrip, comap_principal, eventually_inf_principal]
+  apply eventually_of_forall
+  rintro z hz
+  simp only [Real.rpow_two]
+  rw [sq_le_one_iff hz.1]
+  exact hz.2
+
+-- Smoothly decreasing function when the real part is bounded eventually ≤ 1
+lemma expterm_eventually_le_one (C : ℝ) (n : ℕ) (hn : 1 ≤ n) : ∀ᶠ (z : ℂ)
+    in (comap Complex.im (cocompact ℝ) ⊓ (comap Complex.re (principal  (Icc 0 1)))),
+    C * abs (exp ((z^(2 : ℕ)-1) * (n : ℝ)⁻¹) ) ≤ 1 := by
+  apply eventually_le_of_tendsto_lt  (zero_lt_one' ℝ) _
+  simp only [abs_exp]
+  rw [ show (fun z : ℂ ↦ C * Real.exp (((z^(2 : ℕ)-1) * (n : ℝ)⁻¹).re))
+    = (fun z : ℂ ↦ C * Real.exp ((z.re^(2 : ℕ) - z.im^(2 : ℕ) - 1) * (n : ℝ)⁻¹)) by
+      ext1 z
+      nth_rewrite 1 [← re_add_im z]
+      simp only [mul_re, sub_re, ofReal_im, mul_zero, tsub_zero, ofReal_re]
+      rw [cpow_nat_cast, sq, mul_re]
+      simp only [re_add_im, one_re, Nat.cast_ofNat, Real.rpow_two, mul_eq_mul_left_iff,
+      Real.exp_eq_exp, mul_eq_mul_right_iff, sub_left_inj, inv_eq_zero, Nat.cast_eq_zero, sq] ]
+  nth_rewrite 2 [← mul_zero C]
+  apply Tendsto.const_mul C _
+  rw [Real.tendsto_exp_comp_nhds_zero]
+  apply Tendsto.atBot_mul_const (show 0 < (n : ℝ)⁻¹ by
+    simp only [inv_pos, Nat.cast_pos, lt_of_lt_of_le zero_lt_one hn])
+  -- x.re ^ 2 - x.im ^ 2 - 1 Tendsto at_bot
+  apply tendsto_atBot_add_const_right _ (-1) _
+  simp only [sub_eq_add_neg]
+  apply tendsto_atBot_add_left_of_ge' _ (1 : ℝ) cocompact_strip.eventually_sq_re_le_one _
+  rw [tendsto_neg_atBot_iff]
+  exact cocompact_strip.tendsto_sq_im_atTop
+
+
+-- The function F is bounded above because f is.
+lemma F_BddAbove (f : ℂ → ℂ) (hB : BddAbove ((abs ∘ f) '' (closedStrip 0 1))) :
+    BddAbove ((abs ∘ F f) '' (closedStrip 0 1)) := by
+  -- Rewriting goal
+  simp only [F, image_congr, comp_apply, map_mul]
+  rw [bddAbove_def] at *
+  cases' hB with B hB
+  -- Using bound
+  use B * ((max 1 ((sSupAbsIm f 0)^ (-(1 : ℝ)))) * max 1 ((sSupAbsIm f 1)^ (-(1 : ℝ))))
+  simp only [smul_eq_mul, map_mul, mem_image, forall_exists_index,
+   and_imp, forall_apply_eq_imp_iff₂]
+  intros z hset
+  specialize hB (abs (f z))
+  specialize hB _
```

Comment:
This again generates multiple goals and should come with bullet points.

Reply:
Done